### PR TITLE
Expose a new CountResources API.

### PR DESCRIPTION
**Pull request by nahumcohen**
**What does this pull request do?**
Add a new API CountResources

**What background context can you provide?**
Expose a rest call that count resources

Where should the reviewer start?
Look at the VariableTest for a good example

How should this be manually tested?
Call the resource/{accountName}/{kind}?count=true&search={query}
The query is optional

### CODE REVIEW

File: test/VariablesTest.cs [hunk 1]
```diff
@@ -42,6 +42,18 @@ public void AddSecretTest()
         }
 
         [Test]
+        public void CountVariablesTest()
+        {
+            string variableUri = $"test:///resources/{TestAccount}/{Constants.KIND_VARIABLE}";
+
+            ClearMocker();
+            Mocker.Mock(new Uri(variableUri + "?count=true&search=dummy"), @"{""count"":0}");
```
Comment (acher82) [inline test/VariablesTest.cs]: I think that test with Count different from 0 and 1 will be more effective

File: test/VariablesTest.cs [hunk 2]
```diff
@@ -42,6 +42,18 @@ public void AddSecretTest()
...
+
+            UInt32 result = Client.CountVariables("dummy");
+            Assert.AreEqual(result, 0);
+        }
+
+                [Test]
```
Comment (acher82) [inline test/VariablesTest.cs]: Unnecessary change

File: test/VariablesTest.cs (hunk 1 above)
Comment (nahumcohen) [reply test/VariablesTest.cs]: Fixed

File: test/VariablesTest.cs (hunk 2 above)
Comment (nahumcohen) [reply test/VariablesTest.cs]: Fixed

**Comment by nahumcohen** [comment]
Add CountResources API
